cpp/include/raft/comms/detail/mpi_comms.hpp

```diff
@@ -104,8 +105,14 @@ constexpr MPI_Op get_mpi_op(const op_t op)
 
 class mpi_comms : public comms_iface {
  public:
-  mpi_comms(MPI_Comm comm, const bool owns_mpi_comm)
-    : owns_mpi_comm_(owns_mpi_comm), mpi_comm_(comm), size_(0), rank_(1), next_request_id_(0)
+  mpi_comms(MPI_Comm comm, const bool owns_mpi_comm, cudaStream_t stream)
```

Comment:
Shouldn't we better use rmm::cuda_stream_view instead of cuda_stream_t?

We're using rmm::cuda_stream_view elsewhere (https://github.com/rapidsai/raft/blob/branch-22.04/cpp/include/raft/handle.hpp#L137) and better be consistent unless there is a strong reason to mix the two.